examples/gitserver/hooks/post-receive

```diff
@@ -7,48 +7,74 @@ set -o pipefail
 function key {
   git config --local --get "${1}"
 }
+
 function addkey {
   git config --local --add "${1}" "${2}"
 }
 
 function detect {
-  if detected=$(key openshift.io.detect); then
-    exit 0
-  fi
-  if ! url=$(key gitserver.self.url); then
-    echo "detect: no self url set"
-    exit 0
+  local repoURL="$1"
+  local repoName="$2"
+  local detectionScript=${DETECTION_SCRIPT:-}
+  # The purpose of using a custom detection script is that users can customize the 
+  # 'detect-language' script to return the image that's appropriate for their needs. 
+  # It is possible to just have new-app do code detection. In that case, just set
+  # the customDetection variable to false
```

Comment:
i'm not convinced we should have this enabled by default since it's going to always match maven projects to wildfly, whereas new-app would match them to either wildfly or EAP depending on which imagestreams the cluster has available.